modules/imgproc/src/smooth.cpp

```diff
@@ -3217,35 +3215,48 @@ class BilateralFilter_8u_Invoker :
                     float sum = 0, wsum = 0;
                     int val0 = sptr[j];
                     k = 0;
-                    #if CV_SSE3
-                    if( haveSSE3 )
+#if CV_SIMD128
+                    if( haveSIMD128 )
                     {
-                        __m128 _val0 = _mm_set1_ps(static_cast<float>(val0));
-                        const __m128 _signMask = _mm_load_ps((const float*)bufSignMask);
+                        v_float32x4 _val0 = v_setall_f32(static_cast<float>(val0));
+                        v_float32x4 vsumw = v_setzero_f32();
+                        v_float32x4 vsumc = v_setzero_f32();
 
                         for( ; k <= maxk - 4; k += 4 )
                         {
-                            __m128 _valF = _mm_set_ps(sptr[j + space_ofs[k+3]], sptr[j + space_ofs[k+2]],
-                                                      sptr[j + space_ofs[k+1]], sptr[j + space_ofs[k]]);
-
-                            __m128 _val = _mm_andnot_ps(_signMask, _mm_sub_ps(_valF, _val0));
-                            _mm_store_si128((__m128i*)buf, _mm_cvtps_epi32(_val));
-
-                            __m128 _cw = _mm_set_ps(color_weight[buf[3]],color_weight[buf[2]],
-                                                    color_weight[buf[1]],color_weight[buf[0]]);
-                            __m128 _sw = _mm_loadu_ps(space_weight+k);
-                            __m128 _w = _mm_mul_ps(_cw, _sw);
-                             _cw = _mm_mul_ps(_w, _valF);
-
-                             _sw = _mm_hadd_ps(_w, _cw);
-                             _sw = _mm_hadd_ps(_sw, _sw);
-                             _mm_storel_pi((__m64*)bufSum, _sw);
-
-                             sum += bufSum[1];
-                             wsum += bufSum[0];
+                            v_float32x4 _valF = v_float32x4(sptr[j + space_ofs[k]],
+                                sptr[j + space_ofs[k + 1]],
+                                sptr[j + space_ofs[k + 3]],
+                                sptr[j + space_ofs[k + 2]]);
+                            v_float32x4 _val = v_abs(_valF - _val0);
+                            v_store(buf, v_round(_val));
+
+                            v_float32x4 _cw = v_float32x4(color_weight[buf[0]],
+                                color_weight[buf[1]],
+                                color_weight[buf[2]],
+                                color_weight[buf[3]]);
+                            v_float32x4 _sw = v_load(space_weight+k);
+                            v_float32x4 _w = _cw * _sw;
+                            _cw = _w * _valF;
+
+                            vsumw += _w;
+                            vsumc += _cw;
                         }
+#if CV_SSE3
```

Comment:
The same, we should also check haveSSE3 to be sure that it's presented on user's CPU.